celeri/hmatrix.py

```diff
@@ -316,7 +319,7 @@ def dot(self, x):
 
         # Step 2) Multiply the direct blocks.
         for i, (obs_node, src_node) in enumerate(self.direct_pairs):
-            x_block = x_tree[src_node.idx_start : src_node.idx_end].ravel()
```

Comment:
Why does .ravel() go away here?  Was it just unnecessary or was it doing something bad?

Reply:
Yeah, it was being `.ravel()`-ed again a couple lines down so this was redundant.